### DISCUSSION TITLE
Use config file in subfolder (flat config)

### DISCUSSION THREAD

**Discussion by fudom** [eslint/eslint#18559] [Q&A] [answered]
In prev version of eslint I used multiple config files per folder. Only to set/override one rule to avoid importing e.g. app `two` in app `one`. Or import from `apps` folder in `shared`. 

Background: The shared folder must not know anything about the apps. And the apps must not know anything about each other. An app can only import from itself and shared folder. I would restrict this in tsconfig, but it seems that this is impossible. Therefore I use eslint to warn the developer.

```plaintext
|- apps/one/.eslintrc
|- apps/two/.eslintrc
|- shared/.eslintrc
|- .eslintrc
```

**shared/.eslintrc** (example)

```json
{
  "rules": {
    "no-restricted-imports": "off",
    "@typescript-eslint/no-restricted-imports": [
      "warn",
      {
        "patterns": [
          {
            "group": ["apps/*"],
            "message": "Import from apps is not allowed!",
            "allowTypeImports": true
          }
        ]
      }
    ]
  }
}
```

Worked well. But how to do this with eslint v9 and the flat config? I tried to migrate this by simply changed the file to `eslint.config.mjs` and `export default [{rules}]`. But does not work. Maybe this feature was dropped.

**Comment by fudom** [comment, marked as answer]
My current solution is doing this in the root config file by adding an object with specific files path. It's ok for me. But for the apps, it would be better if they have their own config file. Otherwise I have to do this with every name in root. I'm not experienced in the new eslint flat config. But it seems to be relative override, which I want. This works for me. But is it possible to have multiple config files like with eslintrc?

```js
[/*...*/, {
  files: ['shared/**/*.ts'],
  rules: {
    'no-restricted-imports': 'off',
    '@typescript-eslint/no-restricted-imports': [
      'warn',
      {
        patterns: [
          {
            "group": ["apps/*"],
            "message": "Import from apps is not allowed!",
            "allowTypeImports": true
          },
        ],
      },
    ],
  },
}]
``` 

Idea: You could create the objects dyncamically with a function. Return the array and spread it to the config. In angular you could iterate through the angular json projects app names and insert path with literal strings.

**Reply by mdjermanovic** [reply]
> But is it possible to have multiple config files like with eslintrc?

No, but we're considering adding support for this use case (https://github.com/eslint/eslint/issues/18385).

**Comment by ezolcem** [comment]
So it is not possible to overwrite root eslint.config.js by placing individual eslint.config.js in subfolders anymore with ESLint 9?

**Reply by error-four-o-four** [reply]
It's not at the moment but it's on the agenda.
https://github.com/eslint/eslint/discussions/18574#discussioncomment-9729092

**Reply by madilekci-avena** [reply]
It is possible now using the `unstable_config_lookup_from_file` flag . [Reference](https://eslint.org/docs/latest/use/configure/configuration-files#experimental-configuration-file-resolution)
